state/accumulator.go

```diff
@@ -317,6 +317,16 @@ func (a *Accumulator) Initialise(roomID string, state []json.RawMessage) (Initia
 	return res, err
 }
 
+type AccumulateResult struct {
+	// TODO: is this redundant---identical to len(TimelineNIDs)?
+	NumNew       int
+	TimelineNIDs []int64
+	// ReloadSnapshot is set to the snapshot ID representing the current state of the room, if this Accumulate has
+	// changed state in a non-incremental fashion. If so, the application should reload its view of this room from the
+	// given snapshot. Otherwise this is 0.
+	ReloadFromSnapshot int64
```

Comment:
This is per https://github.com/matrix-org/sliding-sync/issues/294#issue-1885953239

Not convinced we need this; API could always just load up the latest snapshot?